src/main/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapper.java

```diff
@@ -9,77 +11,104 @@
 import it.gov.pagopa.payhub.auth.utils.Constants;
 import it.gov.pagopa.payhub.dto.generated.UserInfo;
 import it.gov.pagopa.payhub.dto.generated.UserOrganizationRoles;
+import it.gov.pagopa.pu.p4pa_organization.dto.generated.Broker;
+import it.gov.pagopa.pu.p4pa_organization.dto.generated.Organization;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.function.Function;
+import java.util.Optional;
 
 @Service
-public class IamUserInfoDTO2UserInfoMapper implements Function<IamUserInfoDTO, UserInfo> {
+public class IamUserInfoDTO2UserInfoMapper {
 
-	public static final String WS_USER_SUFFIX = "-WS_USER";
+    public static final String WS_USER_SUFFIX = "-WS_USER";
+    private final UsersRepository usersRepository;
+    private final OperatorsRepository operatorsRepository;
+    private final OrganizationSearchClient organizationSearchClient;
+    private final boolean organizationAccessMode;
 
-	private final UsersRepository usersRepository;
-	private final OperatorsRepository operatorsRepository;
+    public IamUserInfoDTO2UserInfoMapper(UsersRepository usersRepository, OperatorsRepository operatorsRepository, OrganizationSearchClient organizationSearchClient, boolean organizationAccessMode) {
```

Comment:
you should still use the @Value annotation to inform spring about what properties to read (ps: in order to let it be more readable, lets split this into multiple lines
```suggestion
    public IamUserInfoDTO2UserInfoMapper(UsersRepository usersRepository, OperatorsRepository operatorsRepository, OrganizationSearchClient organizationSearchClient, @Value(...) boolean organizationAccessMode) {
```